packages/flutter_tools/lib/src/commands/screenshot.dart

```diff
@@ -65,9 +65,10 @@ class ScreenshotCommand extends FlutterCommand {
 
   Device device;
 
-  static void validateOptions(String screenshotType, Device device, String observatoryUri) {
+  Future<void> _validateOptions(String screenshotType, String observatoryUri) async {
     switch (screenshotType) {
       case _kDeviceType:
+        device = await findTargetDevice();
         if (device == null) {
           throwToolExit('Must have a connected device for screenshot type $screenshotType');
         }
```

Comment:
while you're at it you could throw if observatoryUri is non-null (see https://github.com/flutter/flutter/issues/80151)

Reply:
Done